src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -388,6 +402,80 @@ public boolean isAppendChangeTimestamp() {
         return this.appendChangeTimestamp != null && this.appendChangeTimestamp.booleanValue();
     }
 
+    public URL getAlternativeUrl() {
+        return alternativeUrl;
+    }
+
+    public boolean isUseHTTPAuth() {
+        return useHTTPAuth;
+    }
+
+    public String getGroupVisibility() {
+        return groupVisibility;
+    }
+
+    public String getRoleVisibility() {
+        return roleVisibility;
+    }
+
+    public boolean isSupportsWikiStyleComment() {
+        return supportsWikiStyleComment;
+    }
+
+    public boolean isRecordScmChanges() {
+        return recordScmChanges;
+    }
+
+    public boolean isUpdateJiraIssueForAllStatus() {
+        return updateJiraIssueForAllStatus;
+    }
+
+    @DataBoundSetter
+    public void setAlternativeUrl(String alternativeUrl) {
+        this.alternativeUrl = toURL(alternativeUrl);
+    }
+
+    @DataBoundSetter
+    public void setUseHTTPAuth(boolean useHTTPAuth) {
+        this.useHTTPAuth = useHTTPAuth;
+    }
+
+    @DataBoundSetter
+    public void setGroupVisibility(String groupVisibility) {
+        this.groupVisibility = Util.fixEmptyAndTrim(groupVisibility);
+    }
+
+    @DataBoundSetter
+    public void setRoleVisibility(String roleVisibility) {
+        this.roleVisibility = Util.fixEmptyAndTrim(roleVisibility);
+    }
+
+    @DataBoundSetter
+    public void setSupportsWikiStyleComment(boolean supportsWikiStyleComment) {
+        this.supportsWikiStyleComment = supportsWikiStyleComment;
+    }
+
+    @DataBoundSetter
+    public void setRecordScmChanges(boolean recordScmChanges) {
+        this.recordScmChanges = recordScmChanges;
+    }
+
+    @DataBoundSetter
+    public void setUserPattern(String userPattern) {
+        this.userPattern = Util.fixEmptyAndTrim(userPattern);
+
+        if (this.userPattern != null) {
```

Comment:
using positive conditions and 'fail fast' approach improves code readability, e.g.
```
if (this.userPattern == null) {
    return null;
}

return Pattern.compile(this.userPattern);
```

Reply:
We are not returning on void